src/pages/node-express/lesson11-a1.mdx

```diff
@@ -0,0 +1,948 @@
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${token}`,
+      },
+    });
+
+    const data = await response.json();
+    let children = [jobsTableHeader];
+
+    if (response.status === 200) {
+      if (data.count === 0) {
+        jobsTable.replaceChildren(...children); // clear this for safety
+      } else {
+        for (let i = 0; i < data.jobs.length; i++) {
+          let rowEntry = document.createElement("tr");
+
+          let editButton = `<td><button type="button" class="editButton" data-id=${data.jobs[i]._id}>edit</button></td>`;
+          let deleteButton = `<td><button type="button" class="deleteButton" data-id=${data.jobs[i]._id}>delete</button></td>`;
+          let rowHTML = `
+            <td>${data.jobs[i].company}</td>
+            <td>${data.jobs[i].position}</td>
+            <td>${data.jobs[i].status}</td>
+            <div>${editButton}${deleteButton}</div>`;
+
+          rowEntry.innerHTML = rowHTML;
+          children.push(rowEntry);
+        }
+        jobsTable.replaceChildren(...children);
+      }
+    } else {
+      message.textContent = data.msg;
+    }
+  } catch (err) {
+    console.log(err);
+    message.textContent = "A communication error occurred.";
+  }
+  enableInput(true);
+  setDiv(jobsDiv);
+};
+```
+
+So, plug this code into `jobs.js` at the appropriate point, and then try the application again.
+You should now be able to see the entries for each job.
+
+However, the edit and delete buttons don't actually work. This is because the click handler
+in `jobs.js` isn't set to look for them yet. We can add a section to the click handler to remedy this.
+
+```js
+      } else if (e.target.classList.contains("editButton")) {
+        message.textContent = "";
+        showAddEdit(e.target.dataset.id);
+      }
+```
+
+The `dataset.id` contains the id of the entry to be edited. That is then passed on to the showAddEdit
+function. So we need to change that function to do something with this parameter.
+
+This function is in `addEdit.js`, and should be changed as follows:
+
+```js
+export const showAddEdit = async (jobId) => {
+  if (!jobId) {
+    company.value = "";
+    position.value = "";
+    status.value = "pending";
+    addingJob.textContent = "add";
+    message.textContent = "";
+
+    setDiv(addEditDiv);
+  } else {
+    enableInput(false);
+
+    try {
+      const response = await fetch(`/api/v1/jobs/${jobId}`, {
+        method: "GET",
+        headers: {
+          "Content-Type": "application/json",
+          Authorization: `Bearer ${token}`,
+        },
+      });
+
+      const data = await response.json();
+      if (response.status === 200) {
+        company.value = data.job.company;
+        position.value = data.job.position;
+        status.value = data.job.status;
+        addingJob.textContent = "update";
+        message.textContent = "";
+        addEditDiv.dataset.id = jobId;
+
+        setDiv(addEditDiv);
+      } else {
```

Comment:
Unlike the existing code; only handling one thing at a time here (200 = update case) with the intention of refactoring later in these instructions to add the (201 = create case) to reduce chance of confusion at this step